service/prometheus/scrapeconfig.go

```diff
@@ -53,116 +56,223 @@ func getTarget(service v1.Service) model.LabelSet {
 func getScrapeConfigs(service v1.Service, certificateDirectory string) []config.ScrapeConfig {
 	clusterID := GetClusterID(service)
 
-	apiServer := config.URL{&url.URL{
-		Scheme: HttpsScheme,
-		Host:   getTargetHost(service),
-	}}
-
-	tlsConfig := config.TLSConfig{
-		CAFile:   key.CAPath(certificateDirectory, clusterID),
-		CertFile: key.CrtPath(certificateDirectory, clusterID),
-		KeyFile:  key.KeyPath(certificateDirectory, clusterID),
+	secureTLSConfig := config.TLSConfig{
+		CAFile:             key.CAPath(certificateDirectory, clusterID),
+		CertFile:           key.CrtPath(certificateDirectory, clusterID),
+		KeyFile:            key.KeyPath(certificateDirectory, clusterID),
+		InsecureSkipVerify: false,
+	}
+	secureHTTPClientConfig := config.HTTPClientConfig{
+		TLSConfig: secureTLSConfig,
+	}
+	insecureTLSConfig := config.TLSConfig{
+		CAFile:             key.CAPath(certificateDirectory, clusterID),
+		CertFile:           key.CrtPath(certificateDirectory, clusterID),
+		KeyFile:            key.KeyPath(certificateDirectory, clusterID),
+		InsecureSkipVerify: true,
+	}
+	insecureHTTPClientConfig := config.HTTPClientConfig{
+		TLSConfig: insecureTLSConfig,
 	}
 
-	clientTlsConfig := tlsConfig
-	clientTlsConfig.InsecureSkipVerify = true
-
-	kubernetesTlsConfig := tlsConfig
-	kubernetesTlsConfig.InsecureSkipVerify = false
-
-	httpClientConfig := config.HTTPClientConfig{
-		TLSConfig: clientTlsConfig,
+	endpointSDConfig := config.ServiceDiscoveryConfig{
+		KubernetesSDConfigs: []*config.KubernetesSDConfig{
+			{
+				APIServer: config.URL{&url.URL{
+					Scheme: HttpsScheme,
+					Host:   getTargetHost(service),
+				}},
+				Role:      config.KubernetesRoleEndpoint,
+				TLSConfig: secureTLSConfig,
+			},
+		},
+	}
+	nodeSDConfig := config.ServiceDiscoveryConfig{
+		KubernetesSDConfigs: []*config.KubernetesSDConfig{
+			{
+				APIServer: config.URL{&url.URL{
+					Scheme: HttpsScheme,
+					Host:   getTargetHost(service),
+				}},
+				Role:      config.KubernetesRoleNode,
+				TLSConfig: secureTLSConfig,
+			},
+		},
 	}
 
-	clusterIDRelabelConfig := &config.RelabelConfig{
+	clusterIDLabelRelabelConfig := &config.RelabelConfig{
 		TargetLabel: ClusterIDLabel,
 		Replacement: clusterID,
-		Action:      config.RelabelReplace,
+	}
+	clusterTypeLabelRelabelConfig := &config.RelabelConfig{
+		TargetLabel: ClusterTypeLabel,
+		Replacement: GuestClusterType,
+	}
+	ipLabelRelabelConfig := &config.RelabelConfig{
+		TargetLabel:  IPLabel,
+		SourceLabels: model.LabelNames{KubernetesSDNodeAddressInternalIPLabel},
+	}
+	roleLabelRelabelConfig := &config.RelabelConfig{
+		TargetLabel:  RoleLabel,
+		SourceLabels: model.LabelNames{KubernetesSDNodeLabelRole},
+	}
+	missingRoleLabelRelabelConfig := &config.RelabelConfig{
+		SourceLabels: model.LabelNames{KubernetesSDNodeLabelRole},
+		Regex:        EmptyRegexp,
+		Replacement:  WorkerRole,
+		TargetLabel:  RoleLabel,
 	}
 
 	scrapeConfigs := []config.ScrapeConfig{
```

Comment:
Not the configuration is essentially the same as https://github.com/giantswarm/g8s-prometheus/issues/137